packages/platforms/mellanox/x86-64/x86-64-mlnx-msn2700/onlp/builds/src/module/src/platform_lib.h

```diff
@@ -0,0 +1,69 @@
+/************************************************************
+ * <bsn.cl fy=2014 v=onl>
+ *
+ *           Copyright 2014 Big Switch Networks, Inc.
+ *
+ * Licensed under the Eclipse Public License, Version 1.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ *        http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.
+ *
+ * </bsn.cl>
+ ************************************************************
+ *
+ *
+ *
+ ***********************************************************/
+#ifndef __PLATFORM_LIB_H__
+#define __PLATFORM_LIB_H__
+
+#include "x86_64_mlnx_msn2700_log.h"
+#include <onlp/fan.h>
+#include <onlp/psu.h>
+
+#define CHASSIS_PSU_COUNT           2
+#define CHASSIS_TOTAL_FAN_COUNT     10
+#define CHASSIS_TOTAL_THERMAL_COUNT 8
+#define CHASSIS_FAN_COUNT     (CHASSIS_TOTAL_FAN_COUNT - CHASSIS_PSU_COUNT)
+#define CHASSIS_THERMAL_COUNT (CHASSIS_TOTAL_THERMAL_COUNT - CHASSIS_PSU_COUNT)
+
+#define PSU1_ID 1
+#define PSU2_ID 2
+
+#define PSU_MODULE_PREFIX "/bsp/module/psu%d_%s"
+#define PSU_POWER_PREFIX "/bsp/power/psu%d_%s"
+#define IDPROM_PATH "/bsp/eeprom/%s%d_info"
+
+int deviceNodeWrite(char *filename, char *buffer, int buf_size, int data_len);
+int deviceNodeWriteInt(char *filename, int value, int data_len);
+int deviceNodeReadBinary(char *filename, char *buffer, int buf_size, int data_len);
+int deviceNodeReadString(char *filename, char *buffer, int buf_size, int data_len);
+
+typedef enum psu_type {
+    PSU_TYPE_UNKNOWN,
+    PSU_TYPE_AC_F2B,
+    PSU_TYPE_AC_B2F
+} psu_type_t;
+
+psu_type_t get_psu_type(int id, char* modelname, int modelname_len);
+
+#define DEBUG_MODE 0
+
+#if (DEBUG_MODE == 1)
```

Comment:
The should be replaced with calls to AIM_LOG_VERBOSE(). 
It is not necessary to have a separate DEBUG enable/disable. The AIM logger has this functionality already. 
(I know this was just copy-paste but I prefer it not be propagated).